Documentation.app/Contents/MacOS/FP concepts.playground/Pages/Rank-N polymorphism.xcplaygroundpage/Contents.swift

```diff
@@ -0,0 +1,155 @@
+*/
+let n: Int = .random(in: 0..<10)
+let randomWordGenerator: () -> String = { (0..<n).forEach { _ in .random(in: "a"..."z") } }
+/*:
+ In the example above we only got one `randomWordGenerator` function per program execution: if we want a different `randomWordGenerator` function with a different `n` we need to re-run the program. Let's create a function that returns a different `randomWordGenerator` function each time we call it. This is an example of a higher-order function, since it returns another function.
+ */
+func makeRandomWordGenerator() -> () -> String {
+    let length: Int = .random(in: 0..<10)
+    return { (0..<length).forEach { _ in .random(in: "a"..."z") } }
+}
+/*:
+ The body of `makeRandomWordGenerator` is basically the same code we had before (we renamed `n` to `length` to stress the fact that `makeRandomWordGenerator` doesn't use any variables from outside its body). But now, each time we call `makeRandomWordGenerator` a different `length` is generated, and thus we get a different `randomWordGenerator` function each time.
+
+ We check this by creating 3 different functions with `makeRandomWordGenerator` and generating one word with each. We should see that each function prints words of different length (unless we are a bit unlucky).
+ */
+for i in 0..<3 {
+    let f = makeRandomWordGenerator()
+    print(f())
+}
+/*:
+ An important thing to realise is that we cannot just generate the random `length` value inside the body of the closure returned from `makeRandomWordGenerator`. If we did so, a `randomWordGenerator` would return strings of different length each time we call it, and this is not what we want. We want each `randomWordGenerator` function to always return strings of the same length.
+
+ This little example demonstrates that closures indeed capture values from their environment. Sine we define `length` inside `makeRandomWordGenerator`,
+ there's no way it can be accessed from outside its body. This means that the closures returned from `makeRandomWordGenerator` indeed carry a copy of `length` with them, because otherwise they couldn't know what length the words they generate are supposed to be.
+
+ ## What exactly is a closure?
+ Now, imagine for a moment that Swift doesn't allow closures to capture values from their environment. Now we can't just simply pass our `randomWordGenerator` functions around, because they cannot carry the value for `length` they need. Is there any way we can achieve a similar behaviour?
+
+ Well, if we have our function inside a struct, it can access all the members of the struct, just like our `randomWordGenerator` body accessed the variable `length`.
+ */
+struct RandomWordGenerator {
+    let length: Int
+
+    func callAsFunction() -> String {
+        (0..<length).forEach { _ in .random(in: "a"..."z") }
+    }
+}
+/*:
+*/
+let anotherRandomWordGenerator = RandomWordGenerator(k: 6)
```

Comment:
I think the name of the property does not match:

```suggestion
let anotherRandomWordGenerator = RandomWordGenerator(length: 6)
```